db_stress_tool/db_stress_listener.h

```diff
@@ -71,22 +71,41 @@ class DbStressListener : public EventListener {
     VerifyFilePath(info.file_path);
     // pretending doing some work here
     RandomSleep();
-    if (FLAGS_read_fault_one_in) {
-      (void)fault_fs_guard->GetAndResetErrorCount();
-      fault_fs_guard->DisableErrorInjection();
+    if (fault_fs_guard) {
+      fault_fs_guard->DisableThreadLocalReadErrorInjection();
+      fault_fs_guard->DisableThreadLocalWriteErrorInjection();
+      fault_fs_guard->DisableThreadLocalMetadataReadErrorInjection();
+      fault_fs_guard->DisableThreadLocalMetadataWriteErrorInjection();
     }
   }
 
   void OnFlushBegin(DB* /*db*/,
                     const FlushJobInfo& /*flush_job_info*/) override {
     RandomSleep();
-    if (FLAGS_read_fault_one_in) {
-      // Hardcoded to inject retryable error as a non-retryable error would put
-      // the DB in read-only mode and then it would crash on the next write.
+    if (fault_fs_guard) {
       fault_fs_guard->SetThreadLocalReadErrorContext(
           static_cast<uint32_t>(FLAGS_seed), FLAGS_read_fault_one_in,
-          true /* retryable */);
-      fault_fs_guard->EnableErrorInjection();
+          FLAGS_inject_error_severity == 1 /* retryable */,
+          FLAGS_inject_error_severity == 2 /* has_data_loss*/);
+      fault_fs_guard->EnableThreadLocalReadErrorInjection();
```

Comment:
We can call Enable..() anyway since `FLAGS_read_fault_one_in` will also control actual injection.